Tools/clinic/clinic.py

```diff
@@ -3500,6 +3520,8 @@ def bad_argument(self, displayname: str, expected: str, *, limited_capi: bool, e
         else:
             if expected_literal:
                 expected = f'"{expected}"'
+            if clinic is not None:
+                clinic.add_include('pycore_modsupport.h', '_PyArg_BadArgument()')
```

Comment:
or perhaps this?

```suggestion
            assert clinic is not None
            clinic.add_include('pycore_modsupport.h', '_PyArg_BadArgument()')
```

Reply:
As explained in my previous comment, I plan to write a follow-up for this code. We should not use the global variable, but pass an argument which *cannot* be None.

Reply:
PR to refactor this code, to avoid `if clinic is not None`: PR #110982.